internal/backend/rest/config.go

```diff
@@ -70,3 +72,28 @@ func prepareURL(s string) string {
 	}
 	return s
 }
+
+var _ restic.ApplyEnvironmenter = &Config{}
+
+// ApplyEnvironment saves values from the environment to the config.
+func (cfg *Config) ApplyEnvironment(prefix string) {
+	username := cfg.URL.User.Username()
+	password, pwd_set := cfg.URL.User.Password()
+
+	if username == "" {
```

Comment:
I'd prefer to either use both `REST_USERNAME` and `REST_PASSWORD` or none. The environment variables should only be used if neither username nor password is configured so far. That would simplify the code and reduce surprises for users.

Reply:
Makes sense, fixed!